python/tutorials/03-matrix-multiplication-all-types.py

```diff
@@ -0,0 +1,532 @@
+    for in_dtype, out_dtype in [('fp16', 'fp16'),
+                                ('bf16', 'bf16'),
+                                ('fp16', 'fp32'),
+                                ('fp32', 'fp32'),
+                                ('fp8e4', 'fp16'),
+                                ('fp8e5', 'fp16'),
+                                ('int8', 'int8'),
+                                ('int8', 'int32')]]
+)
+def test_correctness(M, N, K, in_dtype, out_dtype):
+    a, a_fp16 = gen_input(M, K, in_dtype, 1, device='cuda')
+    b, b_fp16 = gen_input(K, N, in_dtype, 2, device='cuda')
+    # Allocates output.
+    tl_out_dtype = name_to_tl_types[out_dtype]
+    c = torch.empty((M, N), device=a.device, dtype=tl_to_torch_types[tl_out_dtype])
+    matmul(a, b, c, activation="")
+    torch_output = torch.matmul(a_fp16, b_fp16)
+    print(f"triton_output={c}")
+    print(f"torch_output={torch_output}")
+    rtol = 0 if torch.version.hip is None else 1e-2
+    if torch.allclose(c.to(torch.float16), torch_output, atol=5e-2, rtol=rtol):
+        print("✅ Triton and Torch match")
+    else:
+        print("❌ Triton and Torch differ")
+        assert torch.allclose(c, torch_output, atol=1e-2, rtol=rtol)
+
+
+# %%
+# Benchmark
+# ---------
+#
+# Square Matrix Performance
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# We can now compare the performance of our kernel against that of cuBLAS. Here we focus on square matrices,
+# but feel free to arrange this script as you wish to benchmark any other matrix shape.
+
+def get_type(provider):
+    res = re.findall(r'\(.*?\)', provider)
+    return res[0][1:-1]
+
+def get_x_vals():
+    x_vals = [(512 * v, 512 * v, 512 * v) for v in range (2, 17)]
+
+    x_vals += [(4864, 4096, 8192), (9728, 8192, 65536)]
+
+    return x_vals
+
+inout_dtype = {
+    'int8': torch.int8,
+    'fp16': torch.float16,
+    'fp32': torch.float32,
+    'bf16': torch.bfloat16,
+    'fp8e4': torch.float16,
+    'fp8e5': torch.float16,
+}
+
+@triton.testing.perf_report(
+    triton.testing.Benchmark(
+        x_names=['M', 'N', 'K'],  # Argument names to use as an x-axis for the plot
+        x_vals = get_x_vals(),
+        line_arg='provider',  # Argument name whose value corresponds to a different line in the plot
+        # Possible values for `line_arg`
+        line_vals=['rocblas', 'triton(fp16)', 'triton(bf16)', 'triton(int8)', 'triton(fp8e4)', 'triton(fp8e5)'],
+        # Label name for the lines
+        line_names=["rocBLAS.Fp16", "Triton.Bf16", "Triton.Fp16", "Triton.Int8", "Triton.Fp8E4", "Triton.Fp8E5"],
+        # Line styles
+        styles=[('green', '-'), ('purple', '-'), ('blue', '-'), ('red', 'dashed'), ('yellow', 'solid'), ('black', 'dotted')],
+        ylabel="TFLOPS",  # Label name for the y-axis
+        plot_name="matmul-performance",  # Name for the plot, used also as a file name for saving the plot.
+        args={},
+    ))
+def benchmark(M, N, K, provider):
+    if provider == "rocblas":
+        in_dtype = "fp16"
```

Comment:
Why not the other dtypes for rocblas?

Reply:
The current generated input does not work with torch.matmul(). 
@scxiao Have you tried other types with torch.matmul()?

Reply:
@zhanglx13 @vgokhale It works for bfloat16, but not fp8 types.

Reply:
OOC, what library did we get 1300 TF (or whatever that big number was) for fp8?

Reply:
@zhanglx13 , let's add bf16 if it works. It is getting more popular and we will probably get asked for perf there soon. But more importantly, bf16 GEMM last I checked with rocblas was faster than with fp16 (due to thermals) but on Triton it was slower. This was a few months ago so not sure now...

Reply:
@vgokhale  I added rocBLAS.bf16
I tested it on MI300 and triton.bf16 has better perf compared to triton.fp16.
The 1300 tflops is reported from Tensile.